### PR TITLE
Add github auth with automation app token

### PR DESCRIPTION
**Pull request by borjalopez-sg**
## Change description

> Auth git using a temporary token from Terraform Automation App.

## Type of change
- [ ] Bug fix (fixes an issue)
- [X] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -183,6 +183,15 @@ jobs:
           repository: StratusGrid/workflow-config
           ref: main
           path: workflow-config
+      - name: Terraform Automation get install token
+        id: get-token
+        uses: getsentry/action-github-app-token@v1
+        with:
+          app_id: ${{ secrets.APP_ID }}
```
Comment (wesleykirklandsg) [inline .github/workflows/pre-commit.yml]: I think this value needs to be updated

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -183,6 +183,15 @@ jobs:
...
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
```
Comment (wesleykirklandsg) [inline .github/workflows/pre-commit.yml]: I think this value needs to be updated

File: .github/workflows/pre-commit.yml [hunk 3]
```diff
@@ -183,6 +183,15 @@ jobs:
           repository: StratusGrid/workflow-config
           ref: main
           path: workflow-config
+      - name: Terraform Automation get install token
+        id: get-token
+        uses: getsentry/action-github-app-token@38a3ce582e170ddfe8789f509597c6944f2292a9
```
Comment (wesleykirklandsg) [inline .github/workflows/pre-commit.yml]: Can you reference the version tag in a comment and add a comment at the top of the file explaining why we're doing commit SHAs please, you can link to that reference you sent me.

**Review by wesleykirklandsg — COMMENTED** [review]
Requesting some comments

**Review by wesleykirklandsg — APPROVED** [review]
Approved

File: .github/workflows/pre-commit.yml (hunk 3 above)
Comment (borjalopez-sg) [reply .github/workflows/pre-commit.yml]: I pined to SHA with commented version.

**Pull request merged**